### PR TITLE
Added total faucet input value to consensus constants

### PR DESCRIPTION
**Pull request by sdbondi**
## Description

The PR adds the total value for the faucet TXOs that will be used to
validate the pruning horizon UTXO state.

Also added a sanity check to the rincewind genesis block test

## Motivation and Context

The total faucet TXO value is required to validate the pruning horizon UTXO set

## How Has This Been Tested?

N/A - tests making use of this value will be added in the upcoming pruned horizon sync validation PR

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/core/src/consensus/consensus_constants.rs [hunk 1]
```diff
@@ -157,6 +163,7 @@ impl ConsensusConstants {
             emission_decay: 0.999_999_560_409_038_5,
             emission_tail: 1 * T,
             min_pow_difficulty: (1.into(), 60_000_000.into()),
+            total_genesis_input_value: 20471490000000 * uT,
```
Comment (CjS77) [inline base_layer/core/src/consensus/consensus_constants.rs]: Could also do `20_471_490 * T`

**Review by CjS77 — DISMISSED** [review]
One nit

File: base_layer/core/src/consensus/consensus_constants.rs (hunk 1 above)
Comment (sdbondi) [reply base_layer/core/src/consensus/consensus_constants.rs]: Ah of course, done

**Comment by sdbondi** [comment]
This is actually not needed - the chain can be validated with the sum of genesis input commitments.

**Pull request closed without merging**